### ISSUE TITLE
Windows npm bin not on PATH

### ISSUE THREAD

**Issue by ospencer** [actions/setup-node#477]
**Description:**
On Windows, it appears that the npm prefix `C:\npm\prefix` is on the PATH instead of the bin folder within the npm prefix, `C:\npm\prefix\bin`. This causes global packages to be unavailable when installed via `npm install --global`.

**Action version:**
3.1.1

**Platform:**
- [ ] Ubuntu
- [ ] macOS
- [x] Windows

**Runner type:**
- [x] Hosted
- [ ] Self-hosted

**Tools version:**

Node 14, 16, and 18, included version of npm

**Repro steps:**  
1. Use actions/setup-node with any Node version
2. Install a command line package with `--global`, like `shx`: `npm install shx --global`
3. Run the command.

I added a step to this CI run to print out all of the paths on the PATH, and you see `C:\npm\prefix` instead of `C:\npm\prefix\bin` (it's the fourth step):
https://github.com/grain-lang/grain/runs/6200569895

**Expected behavior:**
The command should be available and `$(npm config get prefix)\bin` should be on the PATH.

**Actual behavior:**
The command is not found.

**Comment by dmitry-shibanov** [comment]
Hello @ospencer. Thank you for your report. I've tried to reproduce the issue using the `npm install shx -g` command but it works as expected through `npm`. It looks like in your case you're using yarn to install global packages, that is why the output from the [yarn global bin](https://classic.yarnpkg.com/en/docs/cli/global) command should be added to the `PATH`.

**Comment by ospencer** [comment]
Hey @dmitry-shibanov, I didn't think that was important because of `yarn config set prefix $(npm config get prefix)` (since that works on other OSes) but yep, the relevant documentation is here: https://docs.npmjs.com/cli/v7/configuring-npm/folders#executables

tl;dr yarn uses `<prefix>\bin` always but npm uses `<prefix>/bin` on Unix and just `<prefix>` on Windows.

Wanted to avoid having to alter the PATH just on Windows, but looks we've got to. 😅 

I'll close this. Thanks for your help!
